### PR TITLE
adds code of conduct

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- adds community code of conduct based on https://www.contributor-covenant.org/version/2/0/code_of_conduct/

### CODE REVIEW

**Review by kotharironak — APPROVED** [review]

File: code_of_conduct.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Hypertrace Code of Conduct
```
Comment (adriancole) [inline code_of_conduct.md:1]: @dnielsen is this the content you wanted here? Just to verify before it is copied around

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a 
+   build.
+2. Update the README.md with details of changes to the interface, this includes new environment 
+   variables, exposed ports, useful file locations and container parameters.
+3. Increase the version numbers in any examples files and the README.md to the new version that this
+   Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/).
+4. You may merge the Pull Request once you have the required approvals. If you face any issues merging the Pull Request, you can request reviewer to merge it. 
+
+## Support Channels
```
Comment (adriancole) [inline CONTRIBUTING.md:18]: when pull request is pulled up, this becomes slightly redundant with the first paragraph.  in any case look at some other project CONTRIBUTING files for inspiration and clarification

**Review by adriancole — COMMENTED** [review]
I would put these files under .github which is a standard location, and use the standard case format (ex CoC is in uppercase) https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file

It is best to get things right here before copying around as we still have a lot of repos

**Comment by JBAhire** [comment]
> I would put these files under .github which is a standard location, and use the standard case format (ex CoC is in uppercase) https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file
> 
> It is best to get things right here before copying around as we still have a lot of repos

for this do we need to create a repo `.github`  as instructed in doc?

**Comment by adriancole** [comment]
sorry I wasn't clear `.github` is a root directory in this repo.

It is true that *GitHub* made a little confusion by making a repo named .github, but this is just an exemplar for what to put in the .github dir. https://github.com/github/.github

**Comment by JBAhire** [comment]
got it @adriancole ! :)

File: code_of_conduct.md (hunk 1 above)
Comment (JBAhire) [reply code_of_conduct.md]: @dnielsen , can you please check COC at https://github.com/hypertrace/hypertrace/blob/adding-contributing/.github/code_of_conduct.md and confirm this is what we want?

**Comment by JBAhire** [comment]
handled by #134

**Pull request closed without merging**